src/scheme/controller/SchemeController.ts

```diff
@@ -0,0 +1,38 @@
+import autobind from "autobind-decorator";
+import {SchemeRepository} from "../SchemeRepository";
+import {HttpError, HttpResponse} from "../../service/HttpResponse";
+import {CreateSchemeCommand} from "../command/CreateSchemeCommand";
+
+@autobind
+export class SchemeController {
+
+    constructor(private readonly repository: SchemeRepository,
+                private readonly schemeCmd: CreateSchemeCommand) {}
+
+    public async post(request: SchemeRequest): Promise<SchemeResponse> {
+        let data: SchemeView = {
+            id: null,
+            name: ""
+        };
+        let links = {};
+
+        const scheme = this.schemeCmd.run(request.name);
```

Comment:
why not just do `const data = await this.schemeCmd.run(request.name);` then you don't need the data variable.